pkg/v11/resource/tiller/tiller.go

```diff
@@ -0,0 +1,168 @@
+package tiller
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/errors/guest"
+	"github.com/giantswarm/helmclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/controller/context/resourcecanceledcontext"
+	"github.com/giantswarm/tenantcluster"
+
+	"github.com/giantswarm/cluster-operator/pkg/cluster"
+	"github.com/giantswarm/cluster-operator/pkg/v10/key"
+)
+
+const (
+	Name = "tillerv11"
+)
+
+// Config represents the configuration used to create a new tiller resource.
+type Config struct {
+	BaseClusterConfig        cluster.Config
+	Logger                   micrologger.Logger
+	Tenant                   tenantcluster.Interface
+	ToClusterGuestConfigFunc func(obj interface{}) (v1alpha1.ClusterGuestConfig, error)
+}
+
+// Resource implements the tiller resource.
+type Resource struct {
+	baseClusterConfig        cluster.Config
+	logger                   micrologger.Logger
+	tenant                   tenantcluster.Interface
+	toClusterGuestConfigFunc func(obj interface{}) (v1alpha1.ClusterGuestConfig, error)
+}
+
+// New creates a new configured tiller resource.
+func New(config Config) (*Resource, error) {
+	if reflect.DeepEqual(config.BaseClusterConfig, cluster.Config{}) {
+		return nil, microerror.Maskf(invalidConfigError, "%T.BaseClusterConfig must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.Tenant == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Tenant must not be empty", config)
+	}
+	if config.ToClusterGuestConfigFunc == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ToClusterGuestConfigFunc must not be empty", config)
+	}
+
+	r := &Resource{
+		baseClusterConfig:        config.BaseClusterConfig,
+		logger:                   config.Logger,
+		tenant:                   config.Tenant,
+		toClusterGuestConfigFunc: config.ToClusterGuestConfigFunc,
+	}
+
+	return r, nil
+}
+
```

Comment:
We call ensure tiller installed on both create and delete. This is so the following resources have a working tiller for making changes.